robotframework_reportportal/listener.py

```diff
@@ -188,6 +190,10 @@ def start_test(self, name, attributes, ts=None):
         :param attributes: Dictionary passed by the Robot Framework
         :param ts:         Timestamp(used by the ResultVisitor)
         """
+        if 'source' not in attributes:
+            # no 'source' parameter at this level for Robot versions < 4
+            attributes = attributes.copy()
+            attributes['source'] = self.current_item.source
```

Comment:
No confidence that the current_item has the source attribute

Reply:
Got it.